configure.ac

```diff
@@ -312,98 +239,6 @@ AC_CHECK_HEADER(execinfo.h,
 	[Define to 1 if you have the <execinfo.h> header file.]),
 )
 
-dnl ---------------------------
-dnl Check for curses (optional)
-dnl ---------------------------
-
-HAVE_CURSES=no
-
-AC_ARG_WITH(curses,
-	AS_HELP_STRING([--with-curses(=DIR)],
-		[use curses (optionally, found in DIR)]
-	)
-)
-
-if test "x${with_curses}" != xno; then
-	if test "x${with_curses}" != xyes; then
-		CURSES_CFLAGS="$CURSES_CFLAGS -I${with_curses}/include"
-		CURSES_LDFLAGS="$CURSES_LDFLAGS -L${with_curses}/lib"
-	fi
-	save_CPPFLAGS="$CPPFLAGS"
-	save_LDFLAGS="$LDFLAGS"
-
-	CPPFLAGS="$CPPFLAGS $CURSES_CFLAGS"
-	LDFLAGS="$LDFLAGS $CURSES_LDFLAGS"
-
-	dnl -----------------------
-	dnl Check for curses header
-	dnl -----------------------
-
-	AC_CHECK_HEADER(curses.h,
-		HAVE_CURSES=maybe,
-		HAVE_CURSES=no
-	)
-
-	dnl -----------------
-	dnl Check for ncurses
-	dnl -----------------
-
-	if test "${HAVE_CURSES}" = maybe; then
-		AC_CHECK_LIB(ncurses, initscr,
-			HAVE_CURSES=ncurses
-			AC_DEFINE(HAVE_CURSES, 1,
-					[Define to value if you have a curses compatible library.])
-			CURSES_LIBS="$CURSES_LIBS -lncurses",
-			HAVE_CURSES=maybe
-		)
-	fi
-
-	dnl ------------------
-	dnl Check for pdcurses
-	dnl ------------------
-
-	if test "${HAVE_CURSES}" = maybe; then
-		AC_CHECK_LIB(pdcurses, initscr,
-			HAVE_CURSES=pdcurses
-			AC_DEFINE(HAVE_CURSES, 1,
-					[Define to value if you have a curses compatible library.])
-			CURSES_LIBS="$CURSES_LIBS -lpdcurses",
-			HAVE_CURSES=maybe
-		)
-	fi
-
-	dnl ------------------------
-	dnl Check for generic curses
-	dnl ------------------------
-
-	if test "${HAVE_CURSES}" = maybe; then
-		AC_CHECK_LIB(curses, initscr,
-			HAVE_CURSES=curses
-			AC_DEFINE(HAVE_CURSES, 1,
-					[Define to value if you have a curses library.])
-			CURSES_LIBS="$CURSES_LIBS -lcurses",
-			HAVE_CURSES=no
-		)
-	fi
-
-	dnl --------------------------------------------
-	dnl For dedicated servers, curses is recommended
-	dnl --------------------------------------------
-
-	if test "${HAVE_CURSES}" = no -a "${with_client}" = no; then
-		AC_MSG_ERROR([Could not find curses, which is strongly recommended for dedicated servers.
-		You should install curses, or ./configure --without-curses to ignore this error.
-		])
-	fi
-
-	CPPFLAGS="$save_CPPFLAGS"
-	LDFLAGS="$save_LDFLAGS"
-fi
-
-AC_SUBST(CURSES_CFLAGS)
-AC_SUBST(CURSES_LDFLAGS)
-AC_SUBST(CURSES_LIBS)
-
```

Comment:
The goal here was to streamline our build configuration, such that a build environment will be required to have all of our "core" dependencies (libCURL, libjpeg, libSDL, libxml). Because we are now able to easily produce a dedicated server binary alongside the client binary, we can provide and distribute that ourselves, rather than asking server admins to compile their own executable. So making these dependencies a requirement is not imposing on anyone but us.